libs/datamodel/core/src/diagnostics/mod.rs

```diff
@@ -0,0 +1,67 @@
+mod collection;
+mod error;
+pub(crate) mod warning;
+
+use colored::Colorize;
+
+use crate::ast::Span;
+pub use collection::*;
+pub use error::DatamodelError;
+pub use warning::DatamodelWarning;
+
+/// Given the datamodel text representation, pretty prints an error or warning, including
+/// the offending portion of the source code, for human-friendly reading.
+#[rustfmt::skip]
+fn pretty_print(f: &mut dyn std::io::Write, file_name: &str, text: &str, span: Span, description: &str) -> std::io::Result<()> {
```

Comment:
Can we maybe move all of this into a separate file? I know it has been in the mod file before. But it doesn't seem to be that important that it should be here.